rewrite-kotlin/src/test/java/org/openrewrite/kotlin/cleanup/MoveLambdaArgumentParenthesesTest.java

```diff
@@ -0,0 +1,135 @@
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+import org.openrewrite.test.TypeValidation;
+
+import static org.openrewrite.kotlin.Assertions.kotlin;
+
+class MoveLambdaArgumentParenthesesTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new MoveLambdaArgumentParentheses())
+                .typeValidationOptions(TypeValidation.none())
+                .expectedCyclesThatMakeChanges(1)
+                .cycles(1);
+    }
+
+    @DocumentExample
+    @Test
+    void removeParenthesesFromLet() {
+        rewriteRun(
+          kotlin(
+            """
+              fun method() {
+                  val foo = 1.let({ it + 1 })
+              }
+              """,
+            """
+              fun method() {
+                  val foo = 1.let { it + 1 }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void removeParenthesesFromFilter() {
+        rewriteRun(
+          kotlin(
+            """
+              fun method() {
+                  val list = listOf(1, 2, 3)
+                  list.filter({ it > 1 })
+              }
+              """,
+            """
+              fun method() {
+                  val list = listOf(1, 2, 3)
+                  list.filter { it > 1 }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void removeParenthesesFromRun() {
+        rewriteRun(
+            kotlin(
+            """
+              fun method() {
+                  run({ print("Hello world") })
+              }
+              """,
+            """
+              fun method() {
+                  run { print("Hello world") }
+              }
+              """
+            )
+        );
+    }
+
+    @Test
+    void removeParentheses() {
+        rewriteRun(
+          kotlin(
+            """
+              fun method() {
+                  val list = listOf(1, 2, 3)
+                  list.filterIndexed({ index, value -> index > 1 && value > 1 })
+              }
+              """,
+                """
+              fun method() {
+                  val list = listOf(1, 2, 3)
+                  list.filterIndexed { index, value -> index > 1 && value > 1 }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void noChangeWhenNotLambda() {
+        rewriteRun(
+                spec -> spec.recipe(new MoveLambdaArgumentParentheses())
+                        .expectedCyclesThatMakeChanges(0),
+            kotlin(
+            """
+              fun method() {
+                  val list = listOf(1, 2, 3)
+                  list.filter(::isEven)
+              }
+              """
+            ),
+            kotlin(
+            """
+              fun method() {
+                  run(::print)
+              }
+              """
+            )
+        );
+    }
+}
```

Comment:
It's probably good to add at least one test that only contains a before block, and already uses the desired `{ ... }` form, to show that we do not make any changes in those cases.